HtmRouter.js

```diff
@@ -1,27 +1,54 @@
+/*global useState */
+// Note that this means you have to push useState into global scope
+// in your code. See https://github.com/ruffin--/preact-router-for-htm for more.
 (function () {
+    function isRegExp(obj) {
+        return Object.prototype.toString.call(obj) === '[object RegExp]';
+    }
+
+    function endsWithViaRegExp(str, re) {
+        var reString = re.source;
+        re = reString.endsWith('$') ? re : new RegExp(reString + '$', re.flags);
+        return re.test(str);
+    }
+
+    function endsWithDeluxe(str, stringOrRegExp) {
+        return isRegExp(stringOrRegExp)
+            ? endsWithViaRegExp(str, stringOrRegExp)
+            : str.endsWith(stringOrRegExp);
+    }
+
     window.HtmRouter = function (props) {
         // =============================================
         // jive stolen from
         // https://github.com/preactjs/preact-router
         // =============================================
         function prevent(e) {
-            if (e.stopImmediatePropagation) e.stopImmediatePropagation();
-            if (e.stopPropagation) e.stopPropagation();
+            if (e.stopImmediatePropagation) {
+                e.stopImmediatePropagation();
+            }
+            if (e.stopPropagation) {
+                e.stopPropagation();
+            }
             e.preventDefault();
             return false;
         }
 
         // Handles both delegated and direct-bound link clicks
         function delegateLinkHandler(e) {
             // ignore events the browser takes care of already:
-            if (e.ctrlKey || e.metaKey || e.altKey || e.shiftKey || e.button) return;
+            if (e.ctrlKey || e.metaKey || e.altKey || e.shiftKey || e.button) {
+                return;
+            }
 
             let t = e.target;
             do {
-                if (t.localName === "a" && t.getAttribute("href")) {
-                    if (t.hasAttribute("data-native") || t.hasAttribute("native")) return;
+                if (t.localName === 'a' && t.getAttribute('href')) {
+                    if (t.hasAttribute('data-native') || t.hasAttribute('native')) {
+                        return;
+                    }
 
-                    var newUrl = t.getAttribute("href");
+                    var newUrl = t.getAttribute('href');
```

Comment:
I'm not a single-quote fan as the default, but I realize this is going into a project that expects that standard, so I'll allow it. ;)